discord/app_commands/tree.py

```diff
@@ -1315,3 +1321,22 @@ async def _call(self, interaction: Interaction[ClientT]) -> None:
         else:
             if not interaction.command_failed:
                 self.client.dispatch('app_command_completion', interaction, command)
+
+    async def get_hash(self, *, guild: Optional[Snowflake] = None) -> str:
+        """|coro|
+
+        Returns a hash for tree's state, either for the global tree
+        if no guild is provided, or the guild specific portions of
+        the tree if a guild is provided.
+
+        This can be used to avoid uneccessary syncing.
+
+        Parameters
+        -----------
+        guild: Optional[:class:`~discord.abc.Snowflake`]
+        """
+        payload = await self._get_payload(guild=guild)
+        # versioning allows the method to be changed in the future
+        # blake2 is the fastest collision resistant hash guaranteed
+        # to be available in the standard library at time of writing.
+        return "v1:" + blake2b(_to_json(payload).encode()).hexdigest()
```

Comment:
1. Wouldn't this be better as (version, bytes) or even just bytes? These hashes don't need to be identifiable online.
2. Do we need a 512bit hash digest? Possibly set the digest size lower.
3. This should use `usedforsecurity=False` to not end up a disabled algorithm by security policies.

Reply:
1. Maybe.
2. Probably not, but it isn't hurting anything as-is either. 
3. `usedforsecurity` doesn't exist in python3.8, but you're right. This is going to require a little change here.